mne/io/meas_info.py

```diff
@@ -41,13 +43,75 @@
     seeg=(FIFF.FIFFV_SEEG_CH, FIFF.FIFFV_COIL_NONE, FIFF.FIFF_UNIT_V),
 )
 
-
 def _summarize_str(st):
     """Aux function"""
     return st[:56][::-1].split(',', 1)[-1][::-1] + ', ...'
 
 
-class Info(dict):
+class _ChannelNameList(collections.Sequence):
+    """A read-only list, used to provide convenient acces to channel names."""
+    def __init__(self, info):
+        self._channels = info['chs']
+
+    def __getitem__(self, index):
+        if isinstance(index, int):
+            return self._channels[index]['ch_name']
+        elif isinstance(index, slice):
+            return list([ch['ch_name'] for ch in self._channels[index]])
+
+    def __len__(self):
+        return len(self._channels)
+
+    def __eq__(self, other):
+        return self.to_list() == other
+
+    def __repr__(self):
+        if len(self) == 0:
+            channels = 'empty'
```

Comment:
nitpick, but why not just an empty string? This might be confused as a channel with the name 'empty' ;)